server/app/models/List.js

```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose');
+
+const options = {
+  timestamps: true,
+};
+
+const ListSchema = mongoose.Schema({
+  name: {
+    type: String,
+    required: [true, 'Please enter a list name'],
+  },
+  coverUrl: {
+    type: String,
+    required: [true, 'Please add a cover URL'],
+  },
+  UserId: {
+    type: mongoose.Schema.ObjectId,
+    ref: 'User',
+    required: [true],
+  },
+  products: {
+    type: [mongoose.Schema.ObjectId],
+    ref: 'Product',
+    required: [true],
+  },
+  options,
+});
+
```

Comment:
We can add a `toJSON()` to remove unnecessary fields and change the `_id` to `id` to bypass eslint's no-underscore-dangle. The reason for this is because historically dangling underscores meant private fields, even though JavaScript has no private fields. There is no performance gain, but it's to show that there is no special meaning to these fields and prevent confusion.
```
// Removes unnecessary properties before converting the document
// to a JSON object
ListSchema.set('toJSON', {
  transform: (document, returnedObject) => {
    returnedObject.id = returnedObject._id.toString();
    delete returnedObject._id;
    delete returnedObject.__v;
    delete returnedObject.createdAt;
    delete returnedObject.updatedAt;
  },
});
```